LBHTenancyAPITest/Test/Controllers/GetAllTenancyDetailsForGivenTenancyRef.cs

```
@@ -0,0 +1,268 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Security.Cryptography.X509Certificates;
+using System.Threading.Tasks;
+using LBHTenancyAPI.Controllers;
+using LBHTenancyAPI.UseCases;
+using LBHTenancyAPITest.Helpers;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace LBHTenancyAPITest.Test.Controllers
+{
+    public class GetAllTenancyDetailsForGivenTenancyRef
+    {
+        [Fact]
+        public async Task WhenGivenTenancyRefThatDoesntExist_Tenancy_ShouldRespondWithNoResults()
+        {
+            var allTenancyDetails = new AllTenancyDetailsStub();
+            var response = await GetAllTenancyDetailsForTenancyRef(allTenancyDetails, "NotHere");
+            Assert.NotNull(response);
+
+            var actualJson = JSONHelper.ResponseJson(response);
+            var expectedJson = JsonConvert.SerializeObject(getEmptyJSON());
+
+
+            Assert.Equal(expectedJson, actualJson);
```

I feel like the controller should return a 404 in this case, and an error response. As an API consumer, there'll be no way of differentiating between a tenancy which doesn't exist and a tenancy which has no valid data. We can also use the 404 rate as a way of noticing bugs, whereas this will currently fail silently.

Maybe:

```json
{
  "error": "This tenancy does not exist"
}
```

---

Good point! Added now.